### ISSUE TITLE
support properties expression in WHERE clause for LOOKUP

### ISSUE THREAD

**Issue by wey-gu** [vesoft-inc/nebula#4848]
**Introduction**

```

(root@nebula) [basketballplayer]> LOOKUP ON player WHERE properties(vertex).name == "Tony Parker" YIELD properties(vertex).name as name
[ERROR (-1009)]: SemanticError: Expression (properties(VERTEX).name=="Tony Parker") not supported yet

```

By intuition, this is reasonable and we could infer that it's equivalent to `player.mane == "Tony Parker"`, also, it's supported in GO to do `WHERE properties($$) == "foo"`.

Plus, those who are building ORM/templating dev job will benefit from this, too.

**Contents**

https://discuss.nebula-graph.com.cn/t/topic/11126/12

**Related work**

**Comment by BOFA1ex** [comment]
The same as FETCH, `WHERE properties(vertex) == 'foo'` or `WHERE player.name == 'foo'` expressions are both supported.

It's necessary to make it(syntax) align.

```cpp
struct LookupContext final : public AstContext {
  bool isEdge{false};
  bool dedup{false};
  bool isEmptyResultSet{false};
  int32_t schemaId{-1};
  Expression* filter{nullptr}; // using ExpressionProps repleace it?
  YieldColumns* yieldExpr{nullptr};
  std::vector<std::string> idxReturnCols;
  std::vector<std::string> idxColNames;
  // order by
};
```

**Comment by wey-gu** [comment]
BTW @BOFA1ex , are you planning open-sourcing your downstream OGM ;)

**Comment by BOFA1ex** [comment]
Exactly, the OGM framework has been bind on ourself internal-biz product.
So i cannot given an answer to u, it's out of my control to decide it planning(about open-source).

But i'm glad to make it nebula-contrib better together :)

**Comment by wey-gu** [comment]
Thanks~~ And whenever you are considering splitting the OGM part and open-sourcing, let us know!

**Comment by BOFA1ex** [comment]
It will consider compatible with 2.6.x~3.x? : upstream

**Comment by BOFA1ex** [comment]
It seems that it has become inactive, any planning as feature on next milestone?
